### DISCUSSION TITLE
Popup IsLightDismissEnabled does not work with MVVM

### DISCUSSION THREAD

**Discussion by Orgbrat** [xamarin/XamarinCommunityToolkit#1501] [Q&A]
I am using the MVVM model in my Xamarin.Forms application and I have been using the Acr.UserDialogs. But since they are now in permanent lockdown, I thought I would replace them with XCT Popups. This has actually worked pretty good as most of my dialogs are called from within the viiewmodels thru my DialogService.  I can send in data or get data back as I need. The only issue I have ran into is that there are times when I want the popup to be modal and light dismiss disabled. I have tried everything I know even hard setting IsLightDismissEnabled=false in the Popup XAML header. But I can click outside the popup and it just closes. I know that I can handle the light dismiss event but I truly did not want the popup to close without the user clicking the button on the popup.  All my viewmodels inherit from a viewmodel base class that has a INavigation property. So when ever the ViewModel is created in the page's constructor for the BindingContext the page Navigation object is assigned and then passed thru to the DialogService for use to show the popups.

Has anyone else ran into this issue using the Popups from the ViewModels?

**Comment by pictos** [comment]
If you create a style setting this property to false, does that work?

**Reply by Orgbrat** [reply]
I don't understand, are you recommending to create a Style in my Global styles file and put a Setter in it for the IsLightDismissEnabled property set to false?  Why is that different or better than setting IsLightDismissEnabled=false in the Popup XAML declaration?

**Reply by pictos** [reply]
Yes, that was my idea. I'm not saying that is better or worse I'm asking if that works. If it works will be a workaround for now (;

**Comment by Orgbrat** [comment]
Well maybe I am doing something wrong but the below style will not let the app compile.

Error	XFC0001	Cannot resolve property "IsLightDismissEnabled" on type "Popup (property missing or missing accessors)".


`    <Style x:Key="XctPopupStyle" TargetType="xct:Popup">
        <Setter Property="IsLightDismissEnabled" Value="false" />
    </Style>`
